### PR TITLE
update all the things

### PR DESCRIPTION
**Pull request by danielroedler**

### CODE REVIEW

File: new-test-case.mdx [hunk 1]
```diff
@@ -0,0 +1,93 @@
+---
+title: Create a new test case
+description: "An overview of the various test case creation methods"
+icon: "arrow-progress"
+---
+
+## prompting for test case generation
```
Comment (Germandrummer92) [inline new-test-case.mdx:7]:
```suggestion
## Prompting for test case generation
```

File: new-test-case.mdx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+We are deploying an `AI agent` to find the interactions leading towards a desired user goal.
+While our `AI agent` findes the steps of your test cases, it informs you about the progress it's making by showing the steps.
```
Comment (Germandrummer92) [inline new-test-case.mdx:10]:
```suggestion
While our `AI agent` finds the steps of your test cases, it informs you about the progress it's making by showing the steps.
```

File: new-test-case.mdx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+<Frame caption="AI agent informs about its progress, screenshot 02/2024">
+  <img
+    src="images/AI-Agent-generating-steps.png"
+    alt="AI agent informs about its progress"
+  />
+</Frame>
+
+### Editing your AI discovery
+
+If the Agent went wrong, you can modify the prompt and decide which steps to keep. All subsequent steps are replaced with the new agent run.
```
Comment (Germandrummer92) [inline new-test-case.mdx:21]:
```suggestion
If the Agent generated wrong steps, you can modify the prompt and decide which steps to keep. All subsequent steps are replaced with the new agent output.
```

File: new-test-case.mdx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  />
+</Frame>
+
+### Chaining your tests
+
+When using the AI agent, you can use a dependency to chain test cases together. A user flow is virtually a chain of test cases. Shorter flows are faster, more specific and more reliable.
+
+This is how you do it:
+
+- Give us a path where the AI agent should start its test case generation process.
```
Comment (Germandrummer92) [inline new-test-case.mdx:36]:
```suggestion
- Give us a path where the AI agent should start its test case generation process, e.g. `/overview/`
```

File: new-test-case.mdx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  <img
+    src="images/Test-case-define-dependency.png"
+    alt="Test case chaining"
+    width="300px"
+  />
+</Frame>
+
+<div class="mt-8" />
+
+### Free prompting best practices - work in progress
```
Comment (Germandrummer92) [inline new-test-case.mdx:49]:
```suggestion
### Free prompting best practices
```

File: new-test-case.mdx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+- Example without a cookie banner: “sign in with username and password using \{credentials\}”
+- Example with a cookie banner: “accept cookies then sign in with username and password using \{credentials\}”
+
+3. These are the limitations so far:
+
+- We can only handle flows which happen inside the app. Cross-app or cross-device flows are not supported, e.g. we do not cover email confirmation flow or a 2FA right now.
+- We cannot handle captchas yet.
+- Our AI agent might get blocked by robot detection on some high traffic sites. Sites / apps in production are more bot-protected than staging / test systems.
+
+## record a test case
```
Comment (Germandrummer92) [inline new-test-case.mdx:63]:
```suggestion
## Record a test case
```

File: new-test-case.mdx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+For more manual control and for cases that the AI model struggles with we are offering the option to put in code directly. Idealy produce your code with Playwright Codegen.
```
Comment (Germandrummer92) [inline new-test-case.mdx:65]:
```suggestion
For more manual control and for cases that the AI model struggles with we are offering the option to enter code directly. You can produce this code with Playwright Codegen.
```

File: new-test-case.mdx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+You can find the commands required to use Playwright Codegen directly in Octomind, ready for copy and paste.
```
Comment (Germandrummer92) [inline new-test-case.mdx:66]:
```suggestion
You can find the commands required to use Playwright Codegen directly in Octomind, ready to be copy and pasted.
```

File: new-test-case.mdx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    src="images/Test-case-recording.png"
+    alt="Test case recording"
+    width="300px"
+  />
+</Frame>
+
+### Hint
+
+- make sure to only record actions you actually want in your test case.
+- we are not supporting all interactions. Clicks, key enteres and visibility assertions, however, are supported.
```
Comment (Germandrummer92) [inline new-test-case.mdx:81]:
```suggestion
- We don't support all interactions, yet. Clicks, entering text and visibility assertions, however, are supported.
```

File: new-test-case.mdx [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+## manual test case creation
```
Comment (Germandrummer92) [inline new-test-case.mdx:83]:
```suggestion
## Manual test case creation
```

File: new-test-case.mdx [hunk 11]
```diff
@@ -0,0 +1,93 @@
+---
+title: Create a new test case
+description: "An overview of the various test case creation methods"
+icon: "arrow-progress"
+---
+
+## prompting for test case generation
+
+We are deploying an `AI agent` to find the interactions leading towards a desired user goal.
```
Comment (RunOrVeith) [inline new-test-case.mdx:9]:
```suggestion
We are deploying an `AI agent` to find the interactions leading towards a desired user flow.
```

File: new-test-case.mdx [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+</Frame>
+
+### Hint
+
+- make sure to only record actions you actually want in your test case.
+- we are not supporting all interactions. Clicks, key enteres and visibility assertions, however, are supported.
+
+## manual test case creation
+
+We also added a section to manually create test cases. You can add any number of interaction or assertions via the + icons.
```
Comment (Germandrummer92) [inline new-test-case.mdx:85]:
```suggestion
You can also manually create test cases. You can add any number of interaction or assertions via the + icons.
```

**Review by Germandrummer92 — APPROVED** [review]

File: new-test-case.mdx (hunk 2 above)
Comment (RunOrVeith) [inline new-test-case.mdx:10]:
```suggestion
While our `AI agent` finds the steps of your test cases, it constantly informs you about its progress.
```

File: new-test-case.mdx [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+    src="images/AI-Agent-generating-steps.png"
+    alt="AI agent informs about its progress"
+  />
+</Frame>
+
+### Editing your AI discovery
+
+If the Agent went wrong, you can modify the prompt and decide which steps to keep. All subsequent steps are replaced with the new agent run.
+
+<Frame caption="Relaunch AI discovery based on your comments, screenshot 02/2024">
```
Comment (RunOrVeith) [inline new-test-case.mdx:23]:
```suggestion
<Frame caption="Relaunch AI discovery, screenshot 02/2024">
```

File: new-test-case.mdx [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+  <img
+    src="images/AI-Agent-recreate-steps.png"
+    alt="Relaunch AI discovery based on comments"
+  />
+</Frame>
+
+### Chaining your tests
+
+When using the AI agent, you can use a dependency to chain test cases together. A user flow is virtually a chain of test cases. Shorter flows are faster, more specific and more reliable.
```
Comment (RunOrVeith) [inline new-test-case.mdx:32]: Not just when using the agent, in any case

File: new-test-case.mdx (hunk 6 above)
Comment (RunOrVeith) [inline new-test-case.mdx]:
```suggestion
## Record a test case
```

File: new-test-case.mdx (hunk 7 above)
Comment (RunOrVeith) [inline new-test-case.mdx]:
```suggestion
For more manual control and for cases that the AI model struggles with, we are offering the option to put in code directly. Ideally, produce your code with Playwright Codegen.
```

File: new-test-case.mdx [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+- We can only handle flows which happen inside the app. Cross-app or cross-device flows are not supported, e.g. we do not cover email confirmation flow or a 2FA right now.
+- We cannot handle captchas yet.
+- Our AI agent might get blocked by robot detection on some high traffic sites. Sites / apps in production are more bot-protected than staging / test systems.
+
+## record a test case
+
+For more manual control and for cases that the AI model struggles with we are offering the option to put in code directly. Idealy produce your code with Playwright Codegen.
+You can find the commands required to use Playwright Codegen directly in Octomind, ready for copy and paste.
+
+Once you finished your recording, just copy and paste the whole code into the code field. It needs to be syntactically correct.
```
Comment (RunOrVeith) [inline new-test-case.mdx:68]:
```suggestion
Once you finished your recording, just copy and paste the whole code into the code field.
```

File: new-test-case.mdx (hunk 9 above)
Comment (RunOrVeith) [inline new-test-case.mdx]:
```suggestion
- we are not supporting all interactions. Clicks, entering text and visibility assertions are supported.
```

File: new-test-case.mdx (hunk 10 above)
Comment (RunOrVeith) [inline new-test-case.mdx]:
```suggestion
## Manual test case creation
```

File: new-test-case.mdx (hunk 12 above)
Comment (RunOrVeith) [inline new-test-case.mdx]:
```suggestion
We also added the option to manually create test cases. You can add any number of interaction or assertions via the `+` icons.
```

**Pull request merged**